bftengine/include/bftengine/SimpleClient.hpp

```diff
@@ -45,6 +45,8 @@ enum ClientMsgFlag : uint8_t {
   HAS_PRE_PROCESSED_REQ = 0x4,
   KEY_EXCHANGE_REQ = 0x8,
   EMPTY_CLIENT_REQ = 0x10,
+  RECONFIG_FLAG_REQ = 0x20,       // Same as WRITE_ONLY_REQ | RECONFIG_FLAG_REQ
```

Comment:
I think it might be a bit misleading stating `WRITE_ONLY_REQ` as it is non-existent. Also, the `FLAG` part is not consistent with names of other flags. What about:

- RECONFIG_WRITE_REQ
- RECONFIG_READ_REQ

What do you think?

Reply:
This enum represents the states of an 8-bit tuple. Each bit can represent 2 states. So there are 2^8 (256) states. 
Now in the current implementation, we have used ClientMsgFlag to represent those states and not an uint8_t.

The states are of 3 types:
1) Level - 0 states -- This has only one element which is represented by 0x0. This represents that LSB is 0 and it corresponds to WRITE ONLY request and it's mentioned here as EMPTY FLAG.
2) Level - 1 states -- those states which have only one bit 1 and the rest are all zeros.
3) Level - n states -- these states which have more than one place 1. e.g. 00100001 -- This represents ORed state of few of the Level - 1 or Level - 0 states. 
As we are using the enum type to represent the flag and the flag is not uint8_t, we have to enumerate that subset of states that we are going to use. We may not have to add all 256, as most of those we are not using.

Now here as 0x1 is READ ONLY state, 0x0 should represent WRITE ONLY state. So any Level - 1 state is an ORed state of Level - 0.
So as level - 1 is a vanilla state, RECONFIG_FLAG_REQ is used, but essentially it's RECONFIG WRITE REQ. This is true for all like KEY_EXCHANGE_REQ is actually KEY_EXCHANGE_WRITE_REQ. If in the future we have to enumerate the KEY_EXCHANGE_READ_REQ (which will be 0x9) then we should add a similar comment around KEY_EXCHANGE_REQ.  
That's why RECONFIG_WRITE_REQ will be written as a level-1 state and its name will be RECONFIG_FLAG_REQ, but RECONFIG_FLAG_REQ with READ is a level - 2 state and it will be named as RECONFIG_READ_ONLY_REQ.

Reply:
Reference to a non-existing flag WRITE_ONLY_REQ confuses me, too.

Reply:
Ok @yuliasherman, I will remove that from the comment. But all the level-1 flags are having WRITE_ONLY_REQ flag embedded in them by default.